src/aggregator/aggregator/entry.go

```diff
@@ -1210,6 +1202,12 @@ func (e *Entry) applyValueRateLimit(numValues int64, m rateLimitEntryMetrics) er
 	return errWriteValueRateLimitExceeded
 }
 
+func (e *Entry) setLastAccessed(category metricCategory) {
+	now := e.nowFn().UnixNano()
+	e.metrics.durationBetweenWrites[category].RecordDuration(time.Duration(now - e.lastAccessNanos.Load()))
+	e.lastAccessNanos.Store(now)
```

Comment:
(optional nit) using `last := e.lastAccessNanos.Swap(now)` will batch the load+store operations

Reply:
oh nice! will do in a follow up

Reply:
https://github.com/m3db/m3/pull/3972